test/integration/submit_pledge_test.rb

```diff
@@ -133,5 +133,32 @@ class SubmitPledgeTest < ActionDispatch::IntegrationTest
 
       assert has_link? 'Submit pledge'
     end
+
+    it 'should dynamically update page' do
```

Comment:
come to think of it, just rolling it up into submit pledge and update patient and etc is a better strategy